airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py

```diff
@@ -201,11 +209,17 @@ def request_body_json(
         stream_slice: Mapping[str, Any] = None,
         next_page_token: Mapping[str, Any] = None,
     ) -> Optional[Mapping]:
+
+        self.config["offset"] = self.config.get("offset", 0)
+        self.config["limit"] = self.config.get("limit", 10000)
```

Comment:
Do we really need to mutate `self.config` dict with default values for `offset`, `limit` in this place?
As I see you do it `instantiate_report_class` function.
Why we need to do it here too ?

Reply:
Removed it